crates/services/txpool/src/service.rs

```diff
@@ -271,6 +275,54 @@ where
                 }
             }
 
+            new_peer_subscribed = self.new_tx_gossip_subscription.next() => {
+                // If we are out of capacity, we will skip this event.
+                let _ = self.heavy_async_processor.spawn({
+                    let shared_state = self.tx_pool_shared_state.clone();
+                    let tx_status_change = shared_state.tx_status_sender.clone();
+                    async move {
+                    if let Some(peer_id) = new_peer_subscribed {
+                        // Gathering txs
+                        let peer_tx_ids = shared_state.p2p.request_tx_ids(peer_id.clone()).await.inspect_err(|e| {
+                            tracing::error!("Failed to gather tx ids from peer {}: {}", &peer_id, e);
+                        }).unwrap_or_default();
+                        if peer_tx_ids.is_empty() {
+                            return;
+                        }
+                        let tx_ids_to_ask = shared_state.txpool.lock().filter_existing_tx_ids(peer_tx_ids);
+                        if tx_ids_to_ask.is_empty() {
+                            return;
+                        }
+                        let txs: Vec<Arc<Transaction>> = shared_state.p2p.request_txs(peer_id.clone(), tx_ids_to_ask).await.inspect_err(|e| {
+                            tracing::error!("Failed to gather tx ids from peer {}: {}", &peer_id, e);
+                        }).unwrap_or_default().into_iter().flatten().map(Arc::new).collect();
+                        if txs.is_empty() {
+                            return;
+                        }
+                        // Verifying them
+                        let current_height = *shared_state.current_height.lock();
+                        let (version, params) = shared_state
+                            .consensus_parameters_provider
+                            .latest_consensus_parameters();
+                        let checked_transactions = check_transactions(
+                            &txs,
+                            current_height,
+                            shared_state.utxo_validation,
+                            params.as_ref(),
+                            &shared_state.gas_price_provider,
+                            shared_state.memory_pool.clone()
+                        ).await.into_iter().flatten().collect::<Vec<_>>();
+
+                        shared_state.txpool.lock().insert(
+                            &tx_status_change,
+                            version,
+                            checked_transactions
+                        );
```

Comment:
I think you can just call `SharedState::insert`

Reply:
Yeah good catch I was duplicating the verif. All the boilerplate and generics makes it very hard to read.